### ISSUE TITLE
EmptyBlock should not remove block if it contains a comment (excepted for synchronized block)

### ISSUE THREAD

**Issue by murdos** [openrewrite/rewrite#1209]
The rule https://rules.sonarsource.com/java/RSPEC-108 has the following exception:
> When a block contains a comment, this block is not considered to be empty unless it is a synchronized block. synchronized blocks are still considered empty even with comments because they can still affect program flow.

```kotlin
    @Test
    fun handleComment(jp: JavaParser) = assertUnchanged(
        jp,
        before = """
            public class A {
                {
                    if ("foo".length() > 3) {
                        // this usually does not happen, and we don't care
                    } else {
                        System.out.println("this");
                    }
                }
            }
        """
    )
```

NB: Currently code is transformed to - note the unexpecting location of the comment:
```java
public class A {
    {
        if ("foo".length() <= 3) {
            System.out.println("this");
            // this usually does not happen, and we don't care
        }
    }
}
```

**Comment by traceyyoshima** [comment]
@murdos thanks for reporting the issue, taking a look!

**Comment by traceyyoshima** [comment]
fixed by [ef074f5](https://github.com/openrewrite/rewrite/commit/ef074f57ad487c07420d9ba85d045c13cb0e3d19)
